### PR TITLE
Book endpoint integration test

### PR DESCRIPTION
**Pull request by hamzachikar**
hello guys can you please see my updates in the BookControllerIntegrationTest #2

### CODE REVIEW

File: src/test/java/com/app/store/BookControllerIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+@ExtendWith(SpringExtension.class)
```
Comment (mothmane) [inline src/test/java/com/app/store/BookControllerIntegrationTest.java:19]: the ExtendWith is now part of the annotation SpringBootTest you can delete it from here

File: src/test/java/com/app/store/BookControllerIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    @LocalServerPort
+    int port;
+    @Autowired
+    TestRestTemplate testRestTemplate;
+
+    @Test
+    void should_create_a_book(){
+        var book=new Book("book",new ArrayList<>());
+         final ResponseEntity<Book> bookResponseEntity=
+                 testRestTemplate.postForEntity("http://localhost:"+port+"/books",book,Book.class);
```
Comment (mothmane) [inline src/test/java/com/app/store/BookControllerIntegrationTest.java:31]: build the uri outside this call in a local variable

File: src/test/java/com/app/store/BookControllerIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class BookControllerIntegrationTest {
+    @LocalServerPort
+    int port;
+    @Autowired
+    TestRestTemplate testRestTemplate;
+
+    @Test
```
Comment (mothmane) [inline src/test/java/com/app/store/BookControllerIntegrationTest.java:27]: separe the given when then for readability

File: src/test/java/com/app/store/BookControllerIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        var book=new Book("book",new ArrayList<>());
+         final ResponseEntity<Book> bookResponseEntity=
+                 testRestTemplate.postForEntity("http://localhost:"+port+"/books",book,Book.class);
+        bookResponseEntity.getStatusCode().is2xxSuccessful();
+        assertThat(bookResponseEntity.getBody()).isNotNull();
+    }
+    @Test
+    void should_return_book_by_id(){
+        var id= UUID.fromString("e27294b9-8402-4f90-81ed-357c7bbc7a70");
+        final ResponseEntity<Book> bookResponseEntity=
```
Comment (mothmane) [inline src/test/java/com/app/store/BookControllerIntegrationTest.java:38]: No need for final keywork

Comment (mothmane) [reply src/test/java/com/app/store/BookControllerIntegrationTest.java:38]: No need for type also

File: src/test/java/com/app/store/BookControllerIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        System.out.println(bookResponseEntity.getBody());
+        assertThat(bookResponseEntity.getBody()).isNotNull();
+    }
+
+    @Test
+    void should_delete_a_book_by_id(){
+        var id= UUID.fromString("e27294b9-8402-4f90-81ed-357c7bbc7a70");
+        Assertions.assertThatNoException().isThrownBy(
+                ()->testRestTemplate.delete("http://localhost:"+port+"/books/"+id.toString())
+        );
```
Comment (mothmane) [inline src/test/java/com/app/store/BookControllerIntegrationTest.java:71]: you should check that the get do not return this book 
checking that no exception is throw is not necesseary that the integration test is working

File: src/test/java/com/app/store/BookControllerIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        assertThat(bookResponseEntity.getBody().getName()).isEqualTo("book");
+    }
+
+    @Test
+    void should_return_List_of_books(){
+        final ResponseEntity<List> bookResponseEntity=
+                testRestTemplate.getForEntity(
+                        "http://localhost:"+port+"/books",List.class);
+        bookResponseEntity.getStatusCode().is2xxSuccessful();
+        System.out.println(bookResponseEntity.getBody());
```
Comment (mothmane) [inline src/test/java/com/app/store/BookControllerIntegrationTest.java:62]: remove this line

**Review by mothmane — CHANGES_REQUESTED** [review]

**Review by ichrak2 — APPROVED** [review]

**Review by mothmane — APPROVED** [review]

**Pull request merged**